### ISSUE TITLE
Fix "the `must_not_suspend` lint is unstable" error in nightly coverage generation

### ISSUE THREAD

**Issue by conradoplg** [ZcashFoundation/zebra#3004]
## Motivation

When generating coverage using nightly rust, compilation fails:

```
error[E0658]: the `must_not_suspend` lint is unstable
   --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/vergen-5.1.16/src/lib.rs:300:5
    |
300 |     deny(break_with_label_and_loop, deref_into_dyn_supertrait, must_not_suspend,)
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #83310 <https://github.com/rust-lang/rust/issues/83310> for more information
    = help: add `#![feature(must_not_suspend)]` to the crate attributes to enable

For more information about this error, try `rustc --explain E0658`.
error: could not compile `vergen` due to previous error
```

Fix it and re-enable the coverage generation in CI:
https://github.com/ZcashFoundation/zebra/actions/workflows/coverage.yml

This seems to be an issue in `vergen`, its CI is failing with [the same error](https://github.com/rustyhorde/vergen/runs/4075634239?check_suite_focus=true)

Edit: seems to be solved in the recent release https://github.com/rustyhorde/vergen/releases/tag/5.1.17

**Comment by teor2345** [comment]
I expect there will be a new vergen release with this fix some time before the next sprint.

**Comment by mpguerra** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Engineering-Team-5fb24d9264a3e8000e666a9e/issues/ZcashFoundation/zebra/3004?planning-poker) with ZenHub @conradoplg @dconnolly @jvff @oxarbitrage @teor2345 @upbqdn

**Comment by teor2345** [comment]
Closed by PR #3010.

I'll turn coverage back on now.
